torch/_dynamo/variables/builder.py

```diff
@@ -2128,10 +2128,19 @@ def _wrap_lazy_constant(
         value: Union[int, float, bool, str],
         wrap_fn: Optional[Callable[[Union[int, float]], VariableTracker]] = None,
     ) -> VariableTracker:
-        """Wrap a primitive constant, deferring guard installation if allowed."""
+        """Wrap a primitive constant, deferring guard installation if allowed.
+
+        If wrap_fn is provided, this value should become a SymNodeVariable
+        (e.g., when specialize_int/float is False). In that case, we should NOT
+        use LazyConstantVariable because:
+        1. SymNodeVariable is not a constant - operations should go through the graph
+        2. LazyConstantVariable would install CONSTANT_MATCH guards on realization,
+           causing unnecessary recompiles
+        """
+        if wrap_fn is not None:
```

Comment:
Nit: Is `wrap_fn` still necessary then? Can the caller just directly call `wrap_fn`?